file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/LazyAttribute.kt

```diff
@@ -1,20 +1,63 @@
 package com.pkware.filesystem.attributecaching
 
-import kotlin.reflect.KProperty
-
 /**
- * Stores a mutable attribute that is lazily calculated via [initializer] whenever the value is `null` and accessed.
+ * Stores a mutable attribute [value] that is lazily calculated via [initializer] whenever the value is `null` and
+ * accessed.
+ *
+ * Can check the initialization status of the attribute via [initialized].
+ *
+ * If the mutable attribute [value] is set to `null` its [initialized] status is set to `false`.
+ *
  */
-internal class LazyAttribute<T>(private val initializer: () -> T?) {
+public class LazyAttribute<T>(private val initializer: () -> T?) {
```

Comment:
it needs to be public to expose initialized, which i need and has been needed for testing purposes

Reply:
I've got a commit here with everything in the class being `internal`, including `initialized`. All tests pass.

We'd never have a need to access `LazyAttribute` outside of the provider, would we? Those are private vars on the AttributeCachingPath.

Reply:
thats internal to this project yes, but I am saying the initialized code was for any third party including our own code to verify what's being cached

thats also why i did not initially want to couple the initialized variables with lazy attributes because the initialized/cached flag would be a public property of the attribute caching path main for debugging purposes but it could be used in other ways

Reply:
the intent was to have the value and lazyValue of the lazyAttribute class be as private as could be, and the only thing that would be used to check status/be somewhat accessible was initialized

Reply:
This isn't the kind of class that a library should be exposing, plain and simple. That's all I have to say on the matter.

Reply:
now that i remember correctly you guys had me originally make attribute Caching Path itself internal, but I dont really believe that makes much sense if we do want the ability to debug it/look into it outside of this project

Reply:
Okay then im down for decoupling things and leaving it as they were, because I want keep initialized status exposed/be able to debug with it when needed

Reply:
- [ ] Make LazyAttribute internal